File: GYM.Controller.Api/DTOs/UserReportDTO.cs
namespace GYM.Controller.Api.DTOs;
public class UserReportDTO
{
    public int UserId { get; set; }
    public string GeneratedAt { get; set; } = DateTime.Now.ToString("yyyy-MM-dd HH:mm");
    

    public int TotalMeasurementsTaken { get; set; }
    public decimal AverageWeight { get; set; }
    public decimal WeightChange { get; set; } // Current weight minus first weight
    public string BestMileRun { get; set; } = string.Empty;

    // Detailed Historic Trend Data Rows
    public List<StatsDTO> History { get; set; } = new();
}

File: GYM.FrontEnd/src/types/UserReportDTO.ts
import { type StatsDTO } from "./StatsDTO";

export interface UserReportDTO {
  userId: number;
  generatedAt: string;
  totalMeasurementsTaken: number;
  averageWeight: number;
  weightChange: number;
  bestMileRun: string;
  history: StatsDTO[];
  Top5BestMileRun: StatsDTO[];
  Top5Strength: StatsDTO[];
}